R/dm.R

```diff
@@ -91,26 +90,24 @@ new_dm <- function(src, tables, data_model) {
       as_tibble()
   }
 
-  new_dm2(src, tables, data_model_tables, keys, references, filter = NULL)
+  new_dm2(tables, data_model_tables, keys, references, filter = NULL)
 }
 
-new_dm2 <- function(src = cdm_get_src(base_dm),
-                    tables = cdm_get_tables(base_dm),
+new_dm2 <- function(tables = cdm_get_tables(base_dm),
                     data_model_tables = cdm_get_data_model_tables(base_dm),
                     pks = cdm_get_data_model_pks(base_dm),
                     fks = cdm_get_data_model_fks(base_dm),
                     filter = cdm_get_filter(base_dm),
                     base_dm) {
-
-  stopifnot(!is.null(src))
   stopifnot(!is.null(tables))
+  if (!all_same_source(tables)) abort_not_same_src()
```

Comment:
This constructor should be fast, and not perform unnecessary checks. This check should be part of `validate_dm()`.

See https://adv-r.hadley.nz/s3.html for principles we should abide by.